rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+
+This results in very long builds for lots of software, as exhibited by the number of years people have been complaining about it.
+
+### Having Hydra redistribute redistributable software, without verifying installation media
+
+This would be slightly simpler to implement, but would not have the benefit of being 100% sure our installation media are free.
+
+### Having Hydra redistribute redistributable software, with a check for the installation media
+
+This is the current RFC.
+
+### Building all software, including unfree non-redistributable software
+
+This is quite obviously illegal, and thus not an option.
+
+### Not having the `runnableOnHydra` field on licenses
+
+This would make it impossible for Hydra to build them as things currently stand:
+Hydra would then risk actually running these packages within builds for other derivations (eg. NixOS tests).
+
+This would thus only be compatible with changes to Hydra, that would allow to tag a package as not allowed to run, but only to redistribute.
+Such a change to Hydra would most likely be pretty invasive, and is thus left as future work.
+
+# Prior art
+[prior-art]: #prior-art
+
+According to [this discussion](https://github.com/NixOS/nixpkgs/issues/83433), the current status quo dates back to the 20.03 release meeting.
+More than four years have passed, and it is likely worth rekindling this discussion, especially now that we actually have a Steering Committee.
+
+Recent exchanges have been happening in [this issue](https://github.com/NixOS/nixpkgs/issues/83884).
+
+# Resolved questions
+
+### How large are the packages Hydra would need to additionally store?
+
+`nix-community`'s Hydra instance can give us approximations.
+Its `unfree-redist-full` channel is currently 215G large, including around 200G of NVidia kernel packages and 15G for all the rest of unfree redistributable software.
+Its `cuda` channel is currently 482G large.
+
+Currently, NixOS' hydra pushes around 2TB per month to S3, with rebuilds taken into account.
+Noteworthy is the fact that these 2TB are of compressed data.
+Hence, the expected increase would not be 700G per rebuild, but something lower than this, which is hard to pre-compute.
+
+Regardless, Hydra should be able to deal pretty well even with a one-time 700G data dump.
+The issues would come only if compression were not good, in addition to rebuilds being frequent enough to significantly increase the amount of data Hydra pushes to S3.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Is the list of installation methods correct?
+I took it from my personal history as well as the NixOS website, but there may be others.
+Also, I may have the wrong job name, as I tried to guess the correct job name from the various links.
+
+Do we need a specific `redistributableWhenPatched` field on the license?
+It feels like this would be a bit too much, and probably `redistributable` would be enough.
+However, we may need to have it still.
```

Comment:
We surely can’t legally patch the majority of `unfreeRedistributable` software. We would have to hope that we have legal advice that `patchelf` doesn’t count and that we don’t need to do more than that for anything we want to redistribute.